pkg/kubectl/cmd/annotate.go

```diff
@@ -190,10 +190,26 @@ func (o AnnotateOptions) RunAnnotate(f cmdutil.Factory, cmd *cobra.Command) erro
 		return err
 	}
 
+	var includeUninitialized bool
+	if o.all {
+		// include the uninitialized objects by default
+		// unless explicitly set --include-uninitialized=false
+		includeUninitialized = true
+	}
+	if o.selector != "" {
```

Comment:
I think it's strange if `kubectl annotate pod -l foo=bar` returns different results from `kubectl annotate pod -l foo=bar --all`. I think the current implementation is good.